table/block_based/block.h

```diff
@@ -228,6 +228,49 @@ class Block {
   DataBlockHashIndex data_block_hash_index_;
 };
 
+// A GlobalSeqnoAppliedKey exposes a key with global sequence number applied
+// if configured with `global_seqno != kDisableGlobalSequenceNumber`. It may
+// hold a user key or an internal key since `format_version>=3` index blocks
+// contain user keys. In case it holds user keys, it must be configured with
+// `global_seqno == kDisableGlobalSequenceNumber`.
+class GlobalSeqnoAppliedKey {
```

Comment:
At first this looks like a (not particularly efficient) data holder in which repeated calls to GetKey() without re-Initialize() would needlessly recompute the data.

But no, it appears we expect contents of key_ to change, and that GetKey() is really UpdateAndGetKey(). Please drop the const illusion of GetKey(), because I find it rather misleading. This object clearly owns data (otherwise, how does GetKey() return a Slice?) that is modified by GetKey, and mutating data you clearly own (based on public interface) should not be const function.

Reply:
Yeah, it makes sense, I will change it. My initial goal (and still long-term goal actually) is to have a `GetKey() const` that returns a `ParsedInternalKey`, which would point into `key_` for the user key, and have a copy of the global seqno/original value type. But we are not there yet as the way `BlockIter` tries to be agnostic to internal vs. user key makes it complicated...